app/Listeners/readRecent.php

```diff
@@ -0,0 +1,45 @@
+<?php
+
+namespace App\Listeners;
+
+use App\Models\Recent;
+use Illuminate\Support\Facades\Log;
+use Illuminate\Support\Facades\Auth;
+use Illuminate\Queue\InteractsWithQueue;
+use Illuminate\Support\Facades\Redirect;
+use Illuminate\Contracts\Queue\ShouldQueue;
+
+class readRecent
+{
+
+    /**
+     * Handle the event.
+     */
+    public function handle(object $event)
+    {
+        
+        $book_id = $event->book_id;
+        $chapter_no = $event->chapter_no;
+        $user_id = Auth::user()->user_ID;
+        $recent = Recent::where('book_id', $book_id)
+            ->where('user_id', $user_id)
+            ->first();
+        
+        if ($recent === null) {
+            // The book is not in the recent list, so create a new entry with the chapter.
+            Recent::create([
+                'book_id' => $book_id,
+                'user_id' => $user_id,
+                'chapter' => $chapter_no,
+            ]);
+        } else {
```

Comment:
incearca sa eviti structuri nested. 

in loc sa faci else { if {} } poti sa faci un elseif.

Dar nici asta nu ti-as recomanda

Poti sa restructurezi in felul urmator 

in primul rand, daca $recent nu e instanta de Recent, poti sa il faci - $recent = Recent::create()

Sau si mai bine, cu firstOrCreate - https://laravel.com/api/10.x/Illuminate/Database/Eloquent/Builder.html#method_createOrFirst

In cazurile astea tu o sa ai mereu un $recent care e instanta de Recent. la care poti sa ii faci update la chapter de fiecare data (chiar daca $recent->chapter === $chapter_no, ca nu conteaza.. )